spa-react/src/components/Chat/Chat.tsx

```diff
@@ -0,0 +1,163 @@
+import { collection, deleteDoc, doc, onSnapshot } from "firebase/firestore";
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { useAppDispatch, useAppSelector } from "src/store/hooks";
+import {
+  getChatId,
+  getMessages,
+  getReceiverEmail,
+  getUserEmail,
+} from "src/store/selectors";
+import {
+  resetAllMessages,
+  setNewMessage,
+  setValue,
+} from "src/store/slices/chatSlice";
+import { db } from "src/utils/firebase";
+import { ReactComponent as MessageImg } from "../../assets/message.svg";
+import { InputChat } from "../InputChat/InputChat";
+import { Message } from "../Message/Message";
+import { MessageWithoutAvatar } from "../Message/MessageWithoutAvatar";
+import { IMessage } from "../Message/types";
+import { ChatsSectionSC } from "../Sidebar/style";
+import {
+  ChatSectionSC,
+  ContainerMessagesSC,
+  ImageContainerSC,
+  TextSC,
+} from "./style";
+
+export const Chat = () => {
+  const { reset } = useForm();
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const chatId = useAppSelector(getChatId);
+  const messages = useAppSelector(getMessages);
```

Comment:
move to hook

Reply:
I mean all functions and data. We don't need all this logic and data in component